### ISSUE TITLE
Comparison operators use just first column, depth slice ,etc

### ISSUE THREAD

**Issue by juangomezduaso** [DavisVaughan/rray#104]
``` r
library(rray)
rray(1:6,3:2) > 2
#> [1] FALSE FALSE  TRUE
array(1:6,3:2) > 2
#>       [,1] [,2]
#> [1,] FALSE TRUE
#> [2,] FALSE TRUE
#> [3,]  TRUE TRUE
```

<sup>Created on 2019-04-19 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>
The other comparison operators behave in the same way, ignoting all but the first 1D vector

**Comment by DavisVaughan** [comment]
Yea I haven't gotten around to this part yet. This is inherited from vctrs, and makes sense over there (you collapse the entire row to 1 value, works nicely for data frames). I think it will have elementwise comparisons like base R, but with broadcasting too. This will be implemented with #76

**Comment by juangomezduaso** [comment]
Oh sorry, I overlooked 76.
 And that is what I suposed it was happening, thanks
